### PR TITLE
[OTX][HOT-FIX] Resolve replacement backbone issue

### PR DESCRIPTION
**Pull request by harimkang**
## Summary
- Resolve replacement backbone issue with build.py
- Resolve mmcls init_cfg isssue

## Test Results (tests/integration/cli/test_cli.py)
image

### CODE REVIEW

**Review by jaegukhyun — DISMISSED** [review]
LGTM, Any test results?

File: otx/cli/tools/build.py [hunk 1]
```diff
@@ -103,11 +103,12 @@ def main():
                 raise ValueError("The selected backbone has inputs that the user must enter.")
             builder.merge_backbone(args.model, args.backbone)
 
-    config_manager.write_data_with_cfg(
-        workspace_dir=args.workspace_root,
-        train_data_roots=args.train_data_roots,
-        val_data_roots=args.val_data_roots,
-    )
+    if args.workspace_root is not None:
```
Comment (sungmanc) [inline otx/cli/tools/build.py]: If a user gives `--backbones` with `--workspace_root`, it still causes the problem.

So, I suggest moving this (line#107-111) into line#90. In this case, we also could move line#81-82 to line#78

Comment (harimkang) [reply otx/cli/tools/build.py]: Thanks for finding the problem with this code.
But your suggestion still generates `data.yaml` in unintended circumstances. 
In my opinion, supplementing the if statement in line 106 looks good.
How about `if not args.workspace_root and config_manager.splitted_dataset:` ?

Comment (harimkang) [reply otx/cli/tools/build.py]: Oh, there is a problem with my suggestion. Let's think about it some more.

Comment (harimkang) [reply otx/cli/tools/build.py]: Could you please check this https://github.com/openvinotoolkit/training_extensions/pull/1584/commits/c396f181c15821ee8c82fbb9daec5da9fa897bf3 ?

Comment (harimkang) [reply otx/cli/tools/build.py]: It works as intended for the edge case you commented.

Comment (sungmanc) [reply otx/cli/tools/build.py]: Looks good :)

**Review by sungmanc — APPROVED** [review]

**Comment by harimkang** [comment]
> LGTM, Any test results?

image
I think this works fine, but lacks test cases (train after build -> X). I will add it later.

**Review by jaegukhyun — APPROVED** [review]
